cli/pcluster/cli_commands/compute_fleet_status_manager.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import time
+from enum import Enum
+
+import boto3
+from boto3.dynamodb.conditions import Attr
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ComputeFleetStatus(Enum):
+    """Represents the status of the cluster compute fleet."""
+
+    STOPPED = "STOPPED"  # Fleet is stopped, partitions are inactive.
+    RUNNING = "RUNNING"  # Fleet is running, partitions are active.
+    STOPPING = "STOPPING"  # clustermgtd is handling the stop request.
+    STARTING = "STARTING"  # clustermgtd is handling the start request.
+    STOP_REQUESTED = "STOP_REQUESTED"  # A request to stop the fleet has been submitted.
+    START_REQUESTED = "START_REQUESTED"  # A request to start the fleet has been submitted.
+
+    def __str__(self):
+        return str(self.value)
+
+    @staticmethod
+    def is_start_in_progress(status):
+        """Return True if start is requested or in progress."""
+        return status in {ComputeFleetStatus.START_REQUESTED, ComputeFleetStatus.STARTING}
+
+    @staticmethod
+    def is_stop_in_progress(status):
+        """Return True if stop is requested or in progress."""
+        return status in {ComputeFleetStatus.STOP_REQUESTED, ComputeFleetStatus.STOPPING}
+
+    @staticmethod
+    def is_stop_status(status):
+        """Return True if status is any of the stop ones."""
+        return status in {ComputeFleetStatus.STOP_REQUESTED, ComputeFleetStatus.STOPPING, ComputeFleetStatus.STOPPED}
+
+    @staticmethod
+    def is_start_status(status):
+        """Return True if status is any of the start ones."""
+        return status in {ComputeFleetStatus.START_REQUESTED, ComputeFleetStatus.STARTING, ComputeFleetStatus.RUNNING}
+
+
+class ComputeFleetStatusManager:
+    """Implement functionalities to retrieve and update the compute fleet status."""
+
+    COMPUTE_FLEET_STATUS_KEY = "COMPUTE_FLEET"
```

Comment:
Why uppercase?

Reply:
just to differentiate between this as a known constant label and a nodename? Since they both share the same Id "column"